### PR TITLE
Find order feature

### LINKED ISSUE #34: Find Order Feature
- Allow users to find orders by cheese type
- Allow users to find/filter orders by their status (i.e. whether completed or not)
- Allow users to find/filter orders by customers

### PR DESCRIPTION
**Pull request by weixue123**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -296,6 +296,30 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
    Use case ends.
 
 
+#### Use case: Search for particular orders
+
+**MSS**
+<<<<<<< HEAD
```
Comment (daniellau88) [inline docs/DeveloperGuide.md]: Do resolve conflicts

File: docs/UserGuide.md [hunk 2]
```diff
@@ -161,6 +161,29 @@ Examples:
 * `findcheese s/unassigned`
 * `findcheese t/Gouda s/assigned`
 
+### Finding orders: `findorder`
+
+Searches for particular orders in CHIM.
+
+<<<<<<< HEAD
```
Comment (daniellau88) [inline docs/UserGuide.md]: Same here

File: src/main/java/seedu/address/model/order/predicates/OrderNamePredicate.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return (other instanceof SingleFieldPredicate)
+                && super.equals(other)
+                && this.customerList.equals(((OrderNamePredicate) other).customerList);
+    }
+
+    private String getPhoneByOrder(Order order) {
```
Comment (daniellau88) [inline src/main/java/seedu/address/model/order/predicates/OrderNamePredicate.java]: this not needed here? seems like a repeat in `OrderPhonePredicate`

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (weixue123) [reply docs/DeveloperGuide.md]: mb, fixed liao

File: src/main/java/seedu/address/model/order/predicates/OrderNamePredicate.java (hunk 3 above)
Comment (weixue123) [reply src/main/java/seedu/address/model/order/predicates/OrderNamePredicate.java]: mb, forgot to change the methods' and variables' names after copy-pasting - fixed

**Review by daniellau88 — APPROVED** [review]
LGTM

**Pull request merged**